### ISSUE TITLE
Packet drop while using mavlink forwarding in px4 mavlink module

### ISSUE THREAD

**Issue by myeabhishek** [PX4/PX4-Autopilot#14287]
Hello Team,

I am using Mavlink forwarding to route mavlink messages between Telem1 and Telem 2 ports. I have observed on sending high-frequency data on telem1 i.e. 400 bytes/second, I am getting about 10% packet drop on telem2. This increases linearly with increase in frequency of data. There is no other interference in the system. I am able to receive 100 percent of the data on other hardwares. This is not a communication/RF link issue. 

I can see that there are alot of parameters in PX4 controlling the message buffer size and data rate. How can I tune these parameters to get best results for my application?

Following are my settings in RCS script:
Telem 1: MAVLINK_F "-r 1200 -f -b 57600
Telem 2: 
 mavlink start -d ${MAVLINK_COMPANION_DEVICE} -b 57600 -m onboard -r 5000 -x -f

**Comment by julianoes** [comment]
Can you provide the output of `mavlink status` for the ok and no-ok case?

**Comment by myeabhishek** [comment]
Hi, 

I have attached the output of mavlink status command for both cases. Thank you for your assistance.

![not_okay_case](https://user-images.githubusercontent.com/24542655/75764213-8c95e280-5d63-11ea-8660-a5221bc3d17d.png)

![okay_case](https://user-images.githubusercontent.com/24542655/75764250-991a3b00-5d63-11ea-87d3-9789266b1ad5.png)

**Comment by julianoes** [comment]
I was expecting to see a `rate mult` of 1 for the working case and a number smaller than 1 for the not ok case. Instead the `rate mult` seems the same for both cases. However, the ping statistics seem bogus but that might not matter.

Have you tried choosing a higher `-r ` rate for the companion side? Basically, the `rate mult` is < 1 if the asked rates of all the messages is too much compared to the available rate (`-r`).

**Comment by myeabhishek** [comment]
Yes, I increased the -r for companion link to 10000. The packet loss increased from 10 % to 45 %. 

Is serial baud rate related to these configurations? I have set 57600 for both ports.

**Comment by myeabhishek** [comment]
Following is the screenshot for mavlink status. I increased the baud rate for telem2 to 921600 with 10000 as my -r value. 

My packet loss has decreased from 45% in last attempt to 15% and rate multiplier is also set to 1 now. 

![inreased_telem2_baudrate_and_data_rate](https://user-images.githubusercontent.com/24542655/75775220-81e54880-5d77-11ea-96ff-05e881d15458.png)

**Comment by Jackiemesser** [comment]
Hey, did you ever find a solution to your problem? I'm having a similar issue

**Comment by USS-ICI** [comment]
So my code works for Ardupilot, but drops packets continually with PX4 Firmware, did ya'll find a fix?

**Comment by julianoes** [comment]
Fix coming #21074
